dev/WindowsAppRuntime_BootstrapDLL/tracelogging.h

```diff
@@ -0,0 +1,159 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#pragma once
+
+#include "pch.h"
+
+#if defined(_XSTRING_) && defined(_STRSAFE_H_INCLUDED_) && defined(WI_VERIFY)
+static std::wstring ConvertPackagVersionToString(PACKAGE_VERSION packageVersion) //uint16_t major, uint16_t minor, uint16_t build, uint16_t revision)
+{
+    wchar_t formattedVersion[5 + 1 + 5 + 1 + 5 + 1 + 5 + 1]{};  // "12345.12345.12345.12345" + null-terminator
+    WI_VERIFY(SUCCEEDED(StringCchPrintfW(formattedVersion, ARRAYSIZE(formattedVersion), L"%hu.%hu.%hu.%hu", packageVersion.Major, packageVersion.Minor, packageVersion.Build, packageVersion.Revision)));
+    return std::wstring(formattedVersion);
+}
+#else
+static std::wstring ConvertPackagVersionToString(PACKAGE_VERSION packageVersion) //uint16_t major, uint16_t minor, uint16_t build, uint16_t revision)
+{
+    std::wstring dummy{ L"0.0.0.0" }; return dummy;
+}
+#endif
+
+bool __stdcall wilResultLoggingThreadCallback(const wil::FailureInfo& failure) noexcept;
+
+GUID& GetLifetimeActivityId() noexcept;
+
+class WindowsAppRuntimeBootstrap_TraceLogger final : public wil::TraceLoggingProvider
+{
+    IMPLEMENT_TRACELOGGING_CLASS(
+        WindowsAppRuntimeBootstrap_TraceLogger,
+        "Microsoft.WindowsAppRuntime.MddBootstrap",
+        // {d71ecd75-2924-589c-16dd-68208d1b4015}
+        (0xd71ecd75, 0x2924, 0x589c, 0x16, 0xdd, 0x68, 0x20, 0x8d, 0x1b, 0x40, 0x15));
+
+public:
+    BEGIN_COMPLIANT_CRITICAL_DATA_ACTIVITY_CLASS(Initialize, PDT_ProductAndServicePerformance);
+    void StartActivity(
+        UINT32 majorMinorVersion,
+        PCWSTR versionTag,
+        PACKAGE_VERSION &minFrameworkPackageVersion,
+        UINT32 initializationCount)
+    {
+        // Set lifetime activity Id that helps in corelating all sub-activities/telemetry from a single Mdd Bootstrap lifetime
+        SetRelatedActivityId(GetLifetimeActivityId());
+
+        TraceLoggingClassWriteStart(Initialize,
+            _GENERIC_PARTB_FIELDS_ENABLED,
+            TraceLoggingValue(majorMinorVersion, "majorMinorVersion"),
+            TraceLoggingValue(!versionTag ? L"": versionTag, "versionTag"),
+            TraceLoggingValue(
+                ConvertPackagVersionToString(minFrameworkPackageVersion).c_str(),
+                "minFrameworkPackageVersion"),
+            TraceLoggingValue(initializationCount, "initializationCount"));
+    }
+    void StopWithResult(
+        const HRESULT hresult,
+        UINT32 initializationCount,
+        UINT16 ddlmFindMethodUsed,
```

Comment:
I would drop the ddlmFindMethodUsed

If anything it's interesting to know the IntegrityLevel and IsAppContainer, but as the former is multiple int values perhaps just a `UINT32 flags` and a telemetry enum

```
enum class BootstrapInitializeFlags : uint32_t
{
    None = 0,
    IsElevated = 0x0001,
    IsMediumIL = 0x0002,
    IsLowIL = 0x0003,
    IsAppContainer = 0x0004,
};
DEFINE_ENUM_FLAG_OPERATORS(BootstrapInitializeFlags)
```

and you can compute it inside the telemetry code via

```
#include <Security.IntegrityLevel.h>

BootstrapInitializeFlags ComputeFlags(HANDLE token = nullptr)
{
    BootstrapInitializeFlags flags{};
    const auto integrityLevel{ Security::IntegrityLevel::GetIntegrityLevel(token) };
    if (integrityLevel >= SECURITY_MANDATORY_HIGH_RID)
    {
        flags |= BootstrapInitializeFlags::IsHighIL;
    }
    else if (integrityLevel == SECURITY_MANDATORY_MEDIUM_RID)
    {
        flags |= BootstrapInitializeFlags::IsMediumIL;
    }
    else if (integrityLevel == SECURITY_MANDATORY_LOW_RID)
    {
        flags |= BootstrapInitializeFlags::IsLowIL;
        if (SUCCCEEDED(LOG_IF_FAILED(wil::get_token_is_app_container_nothrow(token, isAppContainer)))
        {
            flags |= BootstrapInitializeFlags::IsAppContainer;
        }
    }
    return flags;
}
```

and in here automagically compute flags and log it

```
const auto flags{ ComputeFlags() };
...
TraceLoggingValue(static_cast<uint32_t>(flags), "Flags")
```


That's equivalent but more useful than knowing AppExtension vs Enumeration